document/src/vespa/document/serialization/annotationdeserializer.cpp

```diff
@@ -102,10 +109,15 @@ void AnnotationDeserializer::readAnnotation(Annotation & annotation) {
     uint32_t type_id = readValue<uint32_t>(_stream);
     uint8_t features = readValue<uint8_t>(_stream);
     uint32_t size = getInt1_2_4Bytes(_stream);
+    if (size > _stream.size()) {
+        LOG(warning, "Annotation of type %u claims size %u > available %zd", type_id, size, _stream.size());
+        fail("Annotation contains SpanNode with bad size");
+        return;
+    }
 
     const AnnotationType *type = _repo.getAnnotationType(type_id);
     if (!type) {
-//        LOG(warning, "Skipping unknown annotation of type %u", type_id);
+        LOG(warning, "Skipping unknown annotation of type %u", type_id);
```

Comment:
Should we use buffered logging for these to avoid swamping the logs if something is producing dodgy input at a high rate?

Reply:
Why not fail-fast instead ?
This indicates mismatch between java and c++ and should just fail fast.

Reply:
this just indicates a different configuration (change in schema file).  So there are lots of unit tests specifically checking that this works as-is.